src/script/interpreter.cpp

```diff
@@ -493,6 +493,26 @@ static bool CheckSignatureEncodingSigHashChoice(const vector<unsigned char> &vch
     {
         return true;
     }
+
+    if (flags & SCRIPT_ENABLE_SCHNORR)
+    {
+        if (vchSig.size() == 64 + (check_sighash == true) ? 1 : 0) // 64 sig length plus 1 sighashtype
```

Comment:
would this simplification work?
`if (vchSig.size() == 64 + (check_sighash ) ? 1 : 0)`

Reply:
as below